### DISCUSSION TITLE
Resizing appears to give different results on the same image

### DISCUSSION THREAD

**Discussion by adhawkins** [ImageMagick/ImageMagick#5855] [Help] [answered]
I'm downloading cover art for an album from the internet, and storing it in the metadata of a FLAC file. The metatblocks in a FLAC file can be at most 16MB, so if the downloaded image is larger than that, I'm resizing it using Magick++:

```
Magick::Blob Blob(Data, Length);
Magick::Image Image(Blob);

Image.resize(Magick::Geometry(1920, 1080));

Magick::Blob NewBlob;
Image.write(&NewBlob);
```

However, if I repeat this process with the same image, I seem to get different results (when doing a byte for byte compare). Is there any way for 'resize' to generate the same image consistently when run on the same source image?

Thanks

Andy

**Comment by snibgo** [comment, marked as answer]
I expect the written blob contains timestamps. You are "doing a byte for byte compare" but you really want to "generate the same image consistently". These are different tests.

**Reply by adhawkins** [reply]
Yes, after analysing the byte differences and consulting the PNG spec, I realised it was the PNG timestamp chunk (tIME) that was changing. I'm now using 'Pixels' to get the image data of the two images and comparing them. Is there a better way of checking if the actual pixel data in two images is identical?

**Reply by urban-warrior** [reply]
Try `identify -verbose 1.png 2.png | grep -i Signature`.  If the signatures are the same, the pixels are identical.  For byte-by-byte, add `-strip` to your command-line to remove the timestamp.

**Reply by adhawkins** [reply]
Thanks, but I'm doing this using the Magick++ API.
